multiqc/modules/bamdst/bamdst.py

```diff
@@ -0,0 +1,419 @@
+""" MultiQC module to parse output from Bamdst """
+
+
+import logging
+from collections import defaultdict
+from typing import Dict, Union
+import csv
+import os
+import fnmatch
+
+import math
+
+from multiqc.modules.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+from multiqc.plots import table, linegraph, bargraph
+from multiqc.utils import config
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+default_exclude_contigs = [
+    "*_alt",
+    "*_decoy",
+    "*_random",
+    "*_fix",
+    "HLA*",
+    "chrUn*",
+    "chrEBV",
+    "chrM",
+]
+
+
+def _read_config():
+    cfg = getattr(config, "bamdst", dict())
+    if not isinstance(cfg, dict):
+        return {}
+
+    cfg["exclude_include_are_set_by_user"] = "exclude_contigs" in cfg or "include_contigs" in cfg
+
+    cfg["include_contigs"] = cfg.get("include_contigs", [])
+    if not isinstance(cfg["include_contigs"], list):
+        cfg["include_contigs"] = []
+
+    cfg["exclude_contigs"] = cfg.get("exclude_contigs", default_exclude_contigs)
+    if not isinstance(cfg["exclude_contigs"], list):
+        cfg["exclude_contigs"] = []
+
+    if cfg["include_contigs"]:
+        log.debug(f"Configuration: would try to include only these contigs: {', '.join(cfg['include_contigs'])}")
+    if cfg["exclude_contigs"]:
+        log.debug(
+            f"Configuration: would exclude these contigs, unless all would be filtered: {', '.join(cfg['exclude_contigs'])}"
+        )
+
+    cutoff = cfg.get("perchrom_fraction_cutoff", 0.0)
+    try:
+        cutoff = float(cutoff)
+    except ValueError:
+        cutoff = 0.0
+    if cutoff != 0.0:
+        log.debug(
+            f"Configuration: coverage cutoff of contigs is {cutoff * 100.0}%, unless everything would be filtered"
+        )
+    cfg["perchrom_fraction_cutoff"] = cutoff
+
+    return cfg
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="Bamdst",
+            anchor="bamdst",
+            href="https://https://github.com/shiquan/bamdst",
+            info="""is a lightweight tool to stat the depth coverage of target regions of bam file(s)""",
+            # doi="", # No DOI
+        )
+
+        self.cfg = _read_config()
+        data_by_sample = dict()
+        data_by_chromosome_by_sample = dict()
+        for f in self.find_log_files("bamdst/coverage"):
+            if data := self._parse_coverage_report(f):
+                data_by_sample[f["s_name"]] = data
+
+                # Parse "chromosomes.report" sitting next to the "coverage.report" file:
+                if os.path.exists(chroms_path := os.path.join(f["root"], "chromosomes.report")):
```

Comment:
Refactoring idea for MultiQC v2 - generally switch from using `os.path` to the `Path` library. `f["root"]` could then be a `Path` object, meaning that this kind of syntax could be simplified.

Instead of:

```python
os.path.join(f["root"], "chromosomes.report")
```

We could simply have:
```python
f["root"] / "chromosomes.report"
```